extensions/sql/contract-definition/store/src/test/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStoreTest.java

```diff
@@ -158,17 +159,11 @@ void saveMany_noneExist() {
     void saveMany_someExist() {
         var definitionsCreated = getContractDefinitions(3);
         sqlContractDefinitionStore.save(definitionsCreated);
-
-        // create some anew, with some modified properties
-        var newDefs = getContractDefinitions(10).stream().peek(cd -> cd.getContractPolicy().getExtensibleProperties().put("somekey", "someval")).collect(Collectors.toList());
-        sqlContractDefinitionStore.save(newDefs);
-
-        //verify that all have the properties
+        sqlContractDefinitionStore.save(getContractDefinitions(10));
 
         var definitionsRetrieved = sqlContractDefinitionStore.findAll();
 
-        assertThat(definitionsRetrieved).isNotNull();
-        assertThat(definitionsRetrieved).allSatisfy(cd -> assertThat(cd.getContractPolicy().getExtensibleProperties()).containsEntry("somekey", "someval"));
+        assertThat(definitionsRetrieved).hasSize(10);
```

Comment:
the `allSatisfy` assertion is actually important, as it makes sure, that all elements have been updated, even the pre-existing ones.

Reply:
I had to remove it since now it's not possible to modify the policy object anymore (the policy object has been replaced by an id). The current assertion verifies that, when you save 3 definitions with id `0, 1, 2` and then you save 10 other definitions `0, 1, 2 ... 9`, the `0, 1, 2` actually got update, so at the end in the store there will be 10 entries, I think it's enough to verify the behavior.

Reply:
verifying that only makes sure that no more than 10 entities exist in total after the second `save`, but it does not assert whether the pre-existing ones have been _skipped_ or _updated_. 
So should the behavior of the store change at some point from upserting to skipping existing ones, the test would still succeed, which seems odd. Personally I generally want stuff to break when I change something.
what do you think about this:
```java
    @Test
    @DisplayName("Save multiple Contract Definitions with some preexisting Definitions")
    void saveMany_someExist() {
        var preexisting = getContractDefinitions(3);
        sqlContractDefinitionStore.save(preexisting);

        var newDefs = getContractDefinitions(10);
        newDefs.forEach(cd -> cd.getSelectorExpression().getCriteria().add(new Criterion("foo", "=", "bar")));
        sqlContractDefinitionStore.save(newDefs);

        var definitionsRetrieved = sqlContractDefinitionStore.findAll();

        assertThat(definitionsRetrieved).hasSize(10);
        var expectedCriterion = new Criterion("foo", "=", "bar");
        assertThat(definitionsRetrieved).allSatisfy(cd -> assertThat(cd.getSelectorExpression().getCriteria()).hasSize(1)
                .allSatisfy(c -> assertThat(c).isEqualTo(expectedCriterion)));
    }
```
[gist is here](https://gist.github.com/paullatzelsperger/426bdf961cd1e5cc52a0b240b55253aa)

Reply:
@paullatzelsperger In my opinion this is over-complicated. I would prefer to have a dedicated test to verify that.
For example, this assertion should be done by `saveOne_alreadyExist_shouldUpdate`:
```
    @Test
    @DisplayName("Save a single Contract Definition that already exists")
    void saveOne_alreadyExist_shouldUpdate() {
        sqlContractDefinitionStore.save(getContractDefinition("id", "policy", "contract"));
        sqlContractDefinitionStore.save(getContractDefinition("id", "updatedAccess", "updatedContract"));

        var result = sqlContractDefinitionStore.findAll();
        
        assertThat(result).hasSize(1).containsExactly(getContractDefinition("id", "updatedAccess", "updatedContract"));
    }
```

PR updated

Reply:
that's fine too, thanks.